server/routes/users.js

```diff
@@ -3,167 +3,473 @@ var express = require('express');
+  var is_admin = req.query.is_admin;
+
+  name_values = {
+    first_name: first_name,
+    last_name: last_name
+  }
+  password_value = {
+    password: password
+  }
+  email_value = {
+    email: email
+  }
+  phone_value = {
+    phone: phone
+  } 
+  university_id_value = {
+    university_id: university_id
+  }
+  is_admin_value = {
+    is_admin: is_admin
+  }
+
+  knex.transaction(async function(trx) { 
+    
+    var query1;
+    var query2;
+    var query3;
+    var query4;
+    var query5;
+    if(first_name && last_name){
+      query1 = knex('swe_user').update(name_values).where({id: user_id});
+    }
+    if(password){
+      query2 = knex('swe_user').update(password_value).where({id: user_id});
+    }
+    if(email){
+      query3 = knex('swe_user').update(email_value).where({id: user_id});
+    }
+    if(university_id){
+      query4 = knex('swe_user').update(university_id_value).where({id: user_id});
+    }
+    if(is_admin){
+      query5 = knex('swe_user').update(is_admin_value).where({id: user_id});
+    }
+
+    if(first_name && last_name){
+      await query1.transacting(trx);
+    }
+    if (password){
+      await query2.transacting(trx);
+    }
+    if (email){
+      await query3.transacting(trx);
+    }
+    if (university_id){
+      await query4.transacting(trx);
+    }
+    if (is_admin){
+      await query5.transacting(trx);
+    }
+    return trx.commit;
+  })
+  .then(result => {
+    res.send(util.message('Successfully updated user'));    
+  })
+  .catch(err => { return next(err) 
+    // If we get here, that means that neither the user insert nor any of the other inserts hav taken place
+  })
+});
+
+// GET a user's past events
 router.get('/:user_id/past', function(req, res, next) {
   const user_id = req.params.user_id;
-  db.query('SELECT event_id FROM event_checkin INNER JOIN event ON event_checkin.event_id = event.fb_id \
-  WHERE event_checkin.user_id = 3 AND event.ends_at < now() UNION SELECT event_id FROM event_registration\
-  INNER JOIN event ON event_registration.event_id = event.fb_id WHERE event_registration.user_id = 3 \
-  AND event.ends_at < now()', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
+  knex('event_checkin')
+    .select('event_id')
+      .innerJoin('event', 'event_checkin.event_id', 'event.fb_id')
+        .where('event_checkin.user_id', 3)
+          .andWhere('event.ends_at' < now())
+            .union([
+              knex.select('event_id')
+                .from('event_registration')
+                  .innerJoin('event', 'event_registration.event_id', 'event.fb_id')
+                    .where('event_checkin.user_id', 3)
+                      .andWhere('event.ends_at' < now())
+
+            ])
+  .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        util.throwError(404, 'No users matching user_id = ' + user_id + ' found');
+      }
+    })
+    .catch(err => { return next(err) 
```

Comment:
This should be one line

.catch(err => { return next(err) })

Reply:
fixed